estate/models/estate_property.py

```diff
@@ -0,0 +1,90 @@
+from datetime import date, timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class Property(models.Model):
+    _name = 'estate.property'
+    _description = "Test description for estate.property model"
+    _order = 'id DESC'
+
+    name = fields.Char(required=True)
+    expected_price = fields.Float(required=True)
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type")
+    state = fields.Selection(
+        selection=[('new', "New"), ('offer_received', "Offer Received"), ('offer_accepted', "Offer accepted"), ('sold', "Sold"), ('cancelled', "Cancelled")],
+        default='new',
+    )
+    description = fields.Text()
+    postcode = fields.Char()
+    selling_price = fields.Float(copy=False, readonly=True)
+    date_availability = fields.Date(copy=False, default=lambda self: date.today() + timedelta(days=90))
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Orientation',
+        selection=[('north', "North"), ('south', "South"), ('east', "East"), ('west', "West")])
+    active = fields.Boolean(default=True)
+    buyer_id = fields.Many2one('res.partner')
+    salesperson_id = fields.Many2one('res.users', copy=False, default=lambda self: self.env.user)
+    tag_ids = fields.Many2many('estate.property.tag')
+    offer_ids = fields.One2many('estate.property.offer', 'property_id')
+    total_area = fields.Float(compute='_compute_total_area')
+    best_price = fields.Float(compute='_compute_best_price')
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)',
+        "The expected price must be strictly positive",
+    )
+    _check_selling_price = models.Constraint(
+        'CHECK (selling_price >= 0)',
+        "The selling price must be positive",
+    )
+
+    @api.depends('garden_area', 'living_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids')
+    def _compute_best_price(self):
+        for record in self:
+            record.best_price = max(o.price for o in record.offer_ids) if record.offer_ids else 0
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = None
+            self.garden_orientation = None
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_check_status(self):
+        for record in self:
+            if record.state not in ['new', 'cancelled']:
+                raise UserError(_("A property can only be deleted if its state is 'New' or 'Cancelled'"))
```

Comment:
We try to have less technical messages for end-users, something like 
```suggestion
                raise UserError(_("A property can only be deleted if it is a new one or if it was cancelled."))
```

Reply:
Also note that `_()` has been mostly replaced by `self.env._()`